docs/guides/: change early EFI boot check

zdykstra opened the pull request:
Neither `hrmpf` or the Debian live CDs mount `efivarfs` by default, so `efibootmgr` fails as an early check. Checking `dmesg` for that string seems to be a reliable check to differentiate BIOS vs EFI booting.

ahesford reviewed (changes requested):
Can you make these `parsed-literal` blocks and bold the user input to be consistent with the other sections where user input and program output are mixed into a common block?

zdykstra commented:
> Can you make these `parsed-literal` blocks and bold the user input to be consistent with the other sections where user input and program output are mixed into a common block?

Moved to a distro-agnostic include and switched to `parsed-literal`. I still need to vet that `dmesg` works without sudo in `hrmpf`.
